Python/optimizer_bytecodes.c

```diff
@@ -370,6 +370,20 @@ dummy_func(void) {
         res = sym_new_type(ctx, &PyUnicode_Type);
     }
 
+    op(_BINARY_OP_SUBSCR_TUPLE_INT, (left, right -- res)) {
+        assert(sym_matches_type(left, &PyTuple_Type));
+        if (sym_is_const(ctx, right)) {
+            assert(PyLong_CheckExact(sym_get_const(ctx, right)));
+            long index = PyLong_AsLong(sym_get_const(ctx, right));
+            assert(index >= 0);
+            assert(index < sym_tuple_length(left));
```

Comment:
I'm not sure these asserts are actually needed. The instruction will deopt when this is not true so perhaps we can remove them?

Reply:
Interesting, we are actually hitting the assertion:
```c
Python/optimizer_cases.c.h:630: optimize_uops: Assertion `index < sym_tuple_length(left)' failed.
```
I guess I need to move the `DEOPT_IF` checks into a separate guard for it to work?

Reply:
Check out the code for `sym_tuple_length`. It can return -1 if the length is not known. So for example if you propagate a PyTuple_Type, but not the length, it will fail.

You need to check that the length is not -1.

Reply:
Got it, thanks! I somehow didn't realize you can know something is a tuple but not know its length..